.github/workflows/main-ci.yml

```
@@ -0,0 +1,707 @@
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_PUBLISH_TOKEN }} # For npmjs.org
+
+# Concurrency for the entire workflow.
+# This might need to be more granular if different job sets need different concurrency rules.
+# For now, using the one from screenshots.yml as a starting point, can be adjusted.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true # More aggressive than "pages", consider if this is too broad.
+                           # If a push comes while a scheduled job runs, it might cancel schedule.
+                           # May need refinement or removal for per-job concurrency.
+
+  cleanup_old_versions:
```

actionlint (1.7.7)

unexpected key "cleanup_old_versions" for "concurrency" section. expected one of "cancel-in-progress", "group"

(syntax-check)